### PR TITLE
DAOS-5632 csum: EC tests for arrays

### PR DESCRIPTION
**Pull request by ryon-jensen**
Test that the checksums for recxs are verified
correctly when EC and checksums is enabled

Signed-off-by: Ryon Jensen <ryon.jensen@intel.com>

Will need https://github.com/daos-stack/daos/pull/3452 to land first

Skip-test: true

### CODE REVIEW

**Review by jolivier23 — APPROVED** [review]

**Review by liuxuezhao — DISMISSED** [review]
thanks, just curious can the added tests pass without PR3452?

**Comment by ryon-jensen** [comment]
> thanks, just curious can the added tests pass without PR3452?

No, this won't land until after PR3452 lands.

**Review by jgmoore-or — DISMISSED** [review]

**Review by jolivier23 — APPROVED** [review]

**Review by jgmoore-or — APPROVED** [review]

**Comment by ryon-jensen** [comment]
This change only added new tests to daos_checksum so there's no way it could cause rebuild tests to fail. Requesting force landing.

**Pull request merged**
